engine/src/engine_api.rs

```diff
@@ -16,41 +16,62 @@ pub enum WriteOperation<'a> {
     /// `Delete` operation
     Delete {
         /// The table name
-        table: &'a str,
+        table: &'static str,
         /// The target key
-        key: &'a [u8],
+        key: Vec<u8>,
```

Comment:
How come you substitute `Vec<u8>` for `&'a [u8]`?  As many data structures in std, say `HashMap`, `HashSet`, or  `Vec`, using a reference to remove a key is enough. There is no need to clone the value here.

Reply:
we need collect `WriteOperation` from different places，and batch write them later，If use references, there will be many lifetime related problems

Reply:
I have pulled this pr and tried to use `&'a [u8]` to replace `Vec<u8>`.  Nothing went wrong. So I think we can still use `&'a [u8]` as before.